omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/ObsByEncounterFragmentController.java

```diff
@@ -70,6 +70,19 @@ public void controller(@SpringBean("patientService") PatientService patientServi
                 concepts.add(concept);
             }
         }
+        node = app.getConfig().get("headers");
+        if (node == null) {
+            throw new IllegalStateException("Missing configuration headers on widget");
+        }
+        List<String> headers = new ArrayList<>();
+        i = node.getElements();
+        while (i.hasNext()) {
+            String header = i.next().getTextValue();
+            if (StringUtils.isNotBlank(header)) {
```

Comment:
Not sure about this use case?  When should it allowed to be blank?  I'd think this should either fail hard, or provide a default.

Reply:
Thanks @mseaton ! I have addressed your observation.